vignettes/notame_normalization_and_feature_selection.qmd

```diff
@@ -0,0 +1,298 @@
+---
+title: "Normalization and feature selection with the notame"
+format: html
+tbl-cap-location: bottom
+editor: visual
+minimal: true
+date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
+vignette: >
+  %\VignetteIndexEntry{Complete end-to-end LC-MS/MS Metabolomic Data analysis}
+  %\VignetteEngine{quarto::html}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+library(knitr)
+library(quarto)
+knitr::opts_knit$set(root.dir = './')
+```
+
+# Introduction
+Herein, we will perform data pretreatment and feature selection using the *notame* R package, which was developed in parallel with the associated protocol article published in the "Metabolomics Data Processing and Data Analysiss—Current Best Practices” special issue of the Metabolites journal (Klåvus et al. 2020) [@klavus_notame_2020]. The main outcome is identifying interesting features for laborious downstream steps relating to biological context, such as annotation and pathway analysis, which fall outside the purview of *notame*. The associated protocol article presents a sequence of steps that is easily adopted by new practitioners. We will not follow the protocol article exactly and instead do an analysis which meshes with the trunk of the workflow.
+
+
+# Setup
+Let's attach the *notame* and load the preprocessed data as a SummarizedExperiment object returned from MSExperiment::quantify().
+
+```{r, message=FALSE, warning=FALSE}
+library(doParallel)
+library(cowplot)
+library(limma)
+library(notame)
+library(SummarizedExperiment)
+
+registerDoParallel(cores = 14)
```

Comment:
and therefore no need for the `doParallel` package

Reply:
Actually, I have a question here @vsuksi - I assume you use `doParallel` parallel processing code within the `notame` package? Also thinking on the Bioconductor submission of your package, I think it would be good to change/switch here to `BiocParallel` parallel processing functions (i.e. `bplapply()`, `bpmapply()` etc). The `BiocParallel` is the *standard* package in Bioconductor to register/configure parallel processing. So, something like the example code above from Phili would then be used across all used packages for the parallel processing. What I found particularly helpful with `BiocParallel` is that it allows to use a variety of different configurations, from `SerialParam` to disable parallel processing, to `SnowParam` (for Windows) `MulticoreParam` (for Unix) to also parameters that allow to configure and use HPC queueing systems like slurm. And there is also a param for `doParallel` too :)

So, maybe, on the longer run, it would make sense to switch the parallel processing code within `notame` to `BiocParallel`?

Reply:
Good to hear and sorry for the delay, I need to practice juggling several things at once! Happily the Bioconductor-version of notame uses BiocParallel instead of doParallel, I will switch to that version and the code chunk above after we're done with some larger changes and submit.  We've been taking some time to make this the first real release, but I'm hoping it is ready in a week or two. 

Do you want all references in the same .bib file or should I include a new one for notame normalization and feature selection?